### ISSUE TITLE
[Phase 4] Photometric Redshift Estimation

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/galaxy-morphology-photoz#5]
# Phase 4: Photometric Redshift Estimation

**Parent Issue:** #1

## Objective

Train regression model for photometric redshift with uncertainty quantification.

## Dependencies

- Phase 1 completed (unified catalog with spec-z)
- Phase 2 completed (features extracted)

### Training Data Preparation
- [ ] Filter to galaxies with spectroscopic redshifts
- [ ] Split train/validation/test (avoid info leakage)
- [ ] Document spectroscopic selection function
- [ ] Analyze spec-z distribution (biased toward bright)

### Model Implementation
- [ ] Implement `src/models/photoz/regressor.py`
- [ ] Create Random Forest regressor with tree variance for uncertainty
- [ ] Alternative: Quantile regression for prediction intervals
- [ ] Add hyperparameter configuration

### Selection Bias Correction
- [ ] Implement `src/models/photoz/bias_correction.py`
- [ ] Compute inverse probability weights based on magnitude
- [ ] Document validity range of photo-z estimates

### Evaluation Metrics
- [ ] Implement photo-z specific metrics:
  - sigma_NMAD: normalized median absolute deviation
  - Outlier fraction: |Δz|/(1+z_spec) > 0.15
  - Bias: mean(Δz/(1+z_spec))
- [ ] Generate photo-z vs spec-z scatter plot
- [ ] Create PIT (probability integral transform) histogram

### Photo-z Metrics
```python
def compute_photoz_metrics(z_spec: np.ndarray, z_phot: np.ndarray) -> dict:
    """Compute standard photo-z quality metrics.

    Args:
        z_spec: True spectroscopic redshifts
        z_phot: Predicted photometric redshifts

    Returns:
        dict with sigma_NMAD, outlier_fraction, bias
    """
    delta_z = (z_phot - z_spec) / (1 + z_spec)

    # Normalized MAD
    sigma_NMAD = 1.48 * np.median(np.abs(delta_z - np.median(delta_z)))

    # Outlier fraction
    outlier_frac = np.mean(np.abs(delta_z) > 0.15)

    # Bias
    bias = np.mean(delta_z)

    return {
        'sigma_NMAD': sigma_NMAD,
        'outlier_fraction': outlier_frac,
        'bias': bias
    }
```

### Random Forest with Uncertainty
```python
class PhotozEstimator:
    def __init__(self, n_estimators: int = 100):
        self.model = RandomForestRegressor(
            n_estimators=n_estimators,
            n_jobs=-1,
            random_state=42
        )

    def predict_with_uncertainty(self, X: np.ndarray) -> tuple:
        """Predict photo-z with uncertainty from tree variance.

        Returns:
            (z_phot, z_sigma): point estimate and uncertainty
        """
        # Get predictions from all trees
        tree_preds = np.array([tree.predict(X) for tree in self.model.estimators_])

        z_phot = np.mean(tree_preds, axis=0)
        z_sigma = np.std(tree_preds, axis=0)

        return z_phot, z_sigma
```

## Verification

```bash
python src/models/photoz/train.py --output models/photoz_rf.pkl
python src/models/photoz/evaluate.py --model models/photoz_rf.pkl
# Expected: sigma_NMAD: 0.025, outlier_frac: 0.03, bias: -0.001
```

## Definition of Done

- [ ] sigma_NMAD < 0.03
- [ ] Outlier fraction < 5%
- [ ] Uncertainty calibration verified (PIT histogram flat)
- [ ] Selection bias documented
- [ ] Photo-z vs spec-z scatter plot saved
